token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -648,7 +782,7 @@ fn process_source_for_transfer(
         source_transfer_amount_lo,
         source_transfer_amount_hi,
     )
-    .ok_or(TokenError::CiphertextArithmeticFailed)?;
+    .ok_or(ProgramError::InvalidInstructionData)?;
```

Comment:
Did you mean to revert all of these errors? Unless there's a good reason, let's stick with the more specific ones

Reply:
This must have been introduced when I was refactoring. I reverted these changes.